service_capacity_modeling/models/org/netflix/ddb.py

```diff
@@ -0,0 +1,584 @@
+            default=0,
+            description="total amount of data stored for backup in gib",
+    )
+    total_annual_backup_cost: float = Field(
+        default=0,
+        description="annual cost to backup the stored data",
+    )
+
+
+class _DataTransferPlan(BaseModel):
+    total_data_transfer_gib: float = Field(
+            default=0,
+            description="amount of data transferred, "
+            "included the global table cross region replication",
+    )
+    total_annual_data_transfer_cost: float = Field(
+        default=0,
+        description="annual cost for data transfer",
+    )
+
+
+def _mean_item_size_bytes(
+    desires: CapacityDesires, extra_model_arguments: Dict[str, Any]
+) -> float:
+    mean_item_size = extra_model_arguments.get("estimated_mean_item_size_bytes", 0)
+    if mean_item_size == 0:
+        mean_item_size = desires.query_pattern.estimated_mean_write_size_bytes.mid
+    return mean_item_size
+
+
+def _get_num_regions(extra_model_arguments: Dict[str, Any]) -> int:
+    return extra_model_arguments.get("number_of_regions", 1)
+
+
+def _get_dynamo_standard(
+    context: RegionContext, extra_model_arguments: Dict[str, Any]
+) -> Service:
+    number_of_regions = _get_num_regions(extra_model_arguments)
+    dynamo_service = (
+        context.services.get("dynamo.standard.global", None)
+        if number_of_regions > 1
+        else context.services.get("dynamo.standard", None)
+    )
+    if not dynamo_service:
+        raise ValueError("DynamoDB Service is not available in context")
+    return dynamo_service
+
+
+def _get_dynamo_backup(context: RegionContext) -> Service:
+    dynamo_backup = context.services.get("dynamo.backup.continuous", None)
+    if not dynamo_backup:
+        raise ValueError("DynamoDB Backup is not available in context")
+    return dynamo_backup
+
+
+def _get_dynamo_transfer(
+    context: RegionContext,
+) -> List[Tuple[float, float]]:
+    transfer_costs = context.services.get("dynamo.transfer", None)
+    if not transfer_costs:
+        raise ValueError(
+            "DynamoDB Transfer cost is not available in context"
+        )
+    return transfer_costs.annual_cost_per_gib
+
+
+def _plan_writes(
+    context: RegionContext,
+    desires: CapacityDesires,
+    extra_model_arguments: Dict[str, Any],
+) -> _WritePlan:
+    # For items up to 1 KB in size,
+    # one WCU can perform one standard write request per second
+
+    mean_item_size = _mean_item_size_bytes(desires, extra_model_arguments)
+
+    rounded_wcus_per_item = math.ceil(max(1.0, mean_item_size / 1024))
+
+    transactional_write_percent: float = extra_model_arguments.get(
+        "transactional_write_percent", 0
+    )
+    assert transactional_write_percent >= 0.0
+    baseline_wcus_non_transactional = (
+        desires.query_pattern.estimated_write_per_second.mid
+        * (1 - transactional_write_percent)
+        * rounded_wcus_per_item
+    )
+    baseline_wcus_transactional = (
```

Comment:
nit: If you have them handle breadcrumbs to AWS docs that explain these various factors (e.g. the 2 wcu per item etc) might be useful.